### ISSUE TITLE
shillelagh error: 'str' object has no attribute 'items' in superset

### ISSUE THREAD

**Issue by weeebdev** [betodealmeida/shillelagh#430]
**Describe the bug**
When I try to connect to json api like `https://www.boredapi.com/api/activity?participants=1`, 
it returns me "shillelagh error: 'str' object has no attribute 'items'"

**To Reproduce**
1. Install superset with shillelagh
2. Try to do an api request

**Expected behavior**
Normal api response

**Screenshots**
<img width="1067" alt="image" src="https://github.com/betodealmeida/shillelagh/assets/48881444/643b6a5b-683b-41e1-ae84-8776b2306561">

**Desktop (please complete the following information):**
 - OS: Mac (Docker Actually)
 - Browser [Arc]

**Additional context**
Here is my config in superset:
```json
{
  "connect_args": {
    "adapters": [
      "genericxmlapi",
      "genericjsonapi",
      "datasetteapi",
      "weatherapi"
    ],
    "adapter_kwargs": {
      "weatherapi": { "api_key": "API_KEY" },
      "genericjsonapi": {
        "request_headers": {
          "Content-type": "application/json"
        }
     }
    
}
```

**Comment by weeebdev** [comment]
Well, it seems that the request is being executed. Something is happening at the response processing.
<img width="725" alt="image" src="https://github.com/betodealmeida/shillelagh/assets/48881444/2313b9ee-ddea-4afe-a51b-46442b1a0191">

**Comment by betodealmeida** [comment]
Taking a look, thanks for the report!

**Comment by betodealmeida** [comment]
@weeebdev I have a fix here: https://github.com/betodealmeida/shillelagh/pull/431

Note that you'll have to specify a different JSONPath to get it to work, since the API returns a single dictionary as the response. You'll need to use this URI: "https://www.boredapi.com/api/activity?participants=1#$"

I'll release a new version as soon as the PR is merged.

**Comment by betodealmeida** [comment]
This is now fixed in 1.2.16.

**Comment by weeebdev** [comment]
@betodealmeida btw, it turned out that I faced another bug when I tried to use rest api provided by https://github.com/hapifhir/hapi-fhir
Apparently, it returns `Content-Type:application/fhir+json;charset=UTF-8` while in the code it directly checks 
```python
return cls.content_type in response.headers.get("content-type", "") #72 line
```
Therefore, I made a quick fix via [regex pattern](https://regex101.com/r/LeYyHa/1). https://github.com/mylifestats/shillelagh/commit/eb8eada6b21a75a07e7f32ee0486fa69bb7e9d2a

Please, consider fixing it.
